--- source-map-generator.js.orig	2017-03-11 22:38:03.750000000 +0300
+++ source-map-generator.js	2017-03-12 00:50:45.687500000 +0300
@@ -97,6 +97,12 @@
       var source = util.getArg(aArgs, 'source', null);
       var name = util.getArg(aArgs, 'name', null);
 
+      // XXX lilliputten 2017.03.11 -- Posix-compatible source & relative(source)
+      source = source.replace(/\\/g,'/');
+      if (this._sourceRoot != null) {
+        source = util.relative(this._sourceRoot.replace(/\\/g,'/'), source);
+      }
+
       if (!this._skipValidation) {
         this._validateMapping(generated, original, source, name);
       }
